versioned/persist/adapter/src/main/java/org/projectnessie/versioned/persist/adapter/SystemPropertiesConfigurer.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.adapter;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to configure instances of {@link DatabaseAdapterConfig} via system properties for
+ * micro-benchmarks and other rather one-off things. Do <em>not</em> use this class in production
```

Comment:
can it live in the `test` module then? in my experience is if its in `main` you can assume someone will use it

Reply:
moved to the nessie-versioned-persist-tests artifact (not in this PR)